test/ruby/test_array.rb

```diff
@@ -1294,6 +1294,13 @@ def test_pack
 =end
   end
 
+  def test_pack_with_buffer
+    # Fixes [Bug #19116]
+    n = [ 65, 66, 67 ]
+    str = "a" * 100
+    assert_equal(n.pack("@3ccc", buffer: str.dup), "aaaABC")
```

Comment:
The order is opposite.
```suggestion
    assert_equal("aaaABC", n.pack("@3ccc", buffer: str.dup))
```